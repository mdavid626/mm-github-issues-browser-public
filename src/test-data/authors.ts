import { Author } from '../types/author';

export const testAuthor1: Author = {
  login: 'testAuthor1',
};
