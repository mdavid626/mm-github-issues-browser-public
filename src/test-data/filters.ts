import { Filters } from '../types/filters';

export const testFilters1: Filters = {
  search: 'test',
  state: 'open',
};
