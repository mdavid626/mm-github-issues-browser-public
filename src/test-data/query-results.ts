export const issueQueryResult = {
  data: {
    repository: {
      id: 'MDEwOlJlcG9zaXRvcnkxMDI3MDI1MA==',
      issue: {
        id: 'I_kwDOAJy2Ks5ePiI3',
        createdAt: '2023-02-12T05:31:50Z',
        title: 'Bug: build args parse error on powsershell',
        state: 'CLOSED',
        number: 26156,
        stateReason: 'COMPLETED',
        url: 'https://github.com/facebook/react/issues/26156',
        body: 'test issue result',
        author: { login: 'weidehai', __typename: 'User' },
        comments: {
          nodes: [],
          pageInfo: {
            hasNextPage: false,
            endCursor: null,
            __typename: 'PageInfo',
          },
          totalCount: 0,
          __typename: 'IssueCommentConnection',
        },
        __typename: 'Issue',
      },
      __typename: 'Repository',
    },
  },
};
