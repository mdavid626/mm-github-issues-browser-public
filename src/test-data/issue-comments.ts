import { IssueComment, IssueComments } from '../types/issue-comment';
import { testAuthor1 } from './authors';

export const testIssueComment1: IssueComment = {
  id: 'IC_kwDOAJy2Ks5Ux_nZ',
  body: 'issue comment 1',
  createdAt: '2023-02-12T09:43:24.641Z',
  author: testAuthor1,
};

export const testIssueComment2: IssueComment = {
  id: 'IC_kwDOAJy2Ks5Ux4u7',
  body: 'issue comment 2',
  createdAt: '2022-01-11T10:42:25.655Z',
  author: testAuthor1,
};

export const testIssueComments1: IssueComments = {
  nodes: [testIssueComment1, testIssueComment2],
  pageInfo: {
    hasNextPage: true,
    endCursor: 'endCursor1',
  },
  totalCount: 2,
};

export const testEmptyIssueComments: IssueComments = {
  nodes: [],
  pageInfo: {
    hasNextPage: false,
    endCursor: 'endCursor1',
  },
  totalCount: 0,
};
