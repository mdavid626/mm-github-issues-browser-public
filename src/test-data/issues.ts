import { Issue, IssueQueryResult, IssuesItem } from '../types/issue';
import { testAuthor1 } from './authors';
import { testIssueComments1 } from './issue-comments';

export const testIssue1: Issue = {
  id: 'I_kwDOAJy2Ks5d65oG',
  createdAt: '2023-02-08T08:50:14Z',
  title: 'Bug: Failed when set homepage in package.json or set PUBLIC_URL',
  state: 'CLOSED',
  number: 26126,
  url: 'https://github.com/facebook/react/issues/26126',
  body: 'test issue1 body',
  author: testAuthor1,
  comments: testIssueComments1,
};

export const testIssuesItem1: IssuesItem = {
  id: 'I_kwDOAJy2Ks5d65oG',
  createdAt: '2023-02-08T08:50:14Z',
  title: 'Bug: Failed when set homepage in package.json or set PUBLIC_URL',
  state: 'CLOSED',
  number: 26126,
  url: 'https://github.com/facebook/react/issues/26126',
  author: testAuthor1,
};

export const testIssueQueryResult1: IssueQueryResult = {
  repository: {
    issue: testIssue1,
  },
};
