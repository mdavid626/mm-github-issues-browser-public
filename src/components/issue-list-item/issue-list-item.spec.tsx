import { cleanup, render } from '@testing-library/react';
import React from 'react';
import { testIssuesItem1 } from '../../test-data/issues';
import IssueStateBadge from '../issue-state-badge/issue-state-badge';
import IssueListItem from './issue-list-item';

jest.mock('../issue-state-badge/issue-state-badge');

describe('issue-list-item', () => {
  afterEach(cleanup);
  afterEach(jest.resetAllMocks);

  it('should render', () => {
    (IssueStateBadge as jest.Mock).mockReturnValue(
      <div>issue-state-badge</div>
    );
    const { asFragment } = render(<IssueListItem issue={testIssuesItem1} />);
    expect(asFragment()).toMatchSnapshot();
    expect(IssueStateBadge).toHaveBeenCalledWith(
      { state: testIssuesItem1.state },
      {}
    );
  });
});
