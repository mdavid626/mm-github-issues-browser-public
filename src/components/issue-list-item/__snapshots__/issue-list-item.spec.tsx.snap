// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`issue-list-item should render 1`] = `
<DocumentFragment>
  <div
    class="IssueListItem"
  >
    <div
      class="IssueListItem-titleAndState"
    >
      <div
        class="IssueListItem-title"
      >
        Bug: Failed when set homepage in package.json or set PUBLIC_URL
      </div>
      <div>
        issue-state-badge
      </div>
    </div>
    <div
      class="IssueListItem-detail"
    >
      <div>
        #26126
      </div>
      <div>
        08.02.2023 09:50
      </div>
      <div>
        testAuthor1
      </div>
    </div>
  </div>
</DocumentFragment>
`;
